### ISSUE TITLE
Copying small files changes them

### ISSUE THREAD

**Issue by nazar-pc** [arpitkh96/AmazeFileManager#330]
### Steps to reproduce
- create text file on Android device with few characters (20 bytes as an example)
- connect to SMB
- copy file through SMB
- check file on SMB server
### Actual result

I see binary file with 61440 bytes (always this number for small files), content is appended to original text file
### Expected result

See original text file with 20 bytes

**Comment by jeweloper** [comment]
I can confirm this bug. The file is manipulated when uploading as well as when downloading.
You can check this by comparing the MD5-Checksum.
Or copying a apk-file on the phone. You will get a parsing error.
Tested on v3.1.1

**Comment by vishal0071** [comment]
Test latest beta (v3.1.2 Beta 3+) and report.

**Comment by nazar-pc** [comment]
I'm on development version of Ubuntu and last days SMB server fails to start unfortunately.
However, from source code it looks fine, so I'm closing issue and will reopen if problem still persists when I'll be able to check it.

**Comment by vishal0071** [comment]
Yes, actually problem is still there. The  buffer rounds up to the nearest 60kbs memory  addresses. I did a temporary fix for just first 60 kbs but the problem is still there. Try copying a 100 kb file and chances are it'll turn up to 120, same with higher size files. Try this yourself, reopen this if comes out to be true. I'll try to workout permanently then.

**Comment by nazar-pc** [comment]
I've copied successfully file with 3 bytes, but 100 kb file was indeed padded with zeroes to 120 kb

**Comment by vishal0071** [comment]
Fixed in latest beta v3.1.2 beta 5, test and report.

**Comment by nazar-pc** [comment]
When copying attached file from samba server to Android device I'm getting 2 messages that application crashed (no visible crash though), file is not copied.

[txt.txt](https://github.com/arpitkh96/AmazeFileManager/files/601755/txt.txt)

**Comment by vishal0071** [comment]
What about normal copy?

**Comment by nazar-pc** [comment]
Copying locally on Android device works fine. Copying from/to samba server causes crash message.

**Comment by nazar-pc** [comment]
With latest update doesn't seem to copy anything at all (trying to copy few MiB image). No crash, no visible progress. It just refreshes files list immediately and nothing.

**Comment by vishal0071** [comment]
What's the source and target for copy?

**Comment by nazar-pc** [comment]
Source is Android (internal storage on Nexus 6P), target is Samba server on Ubuntu 17.04.

**Comment by vishal0071** [comment]
Couldn't test on samba, but normal copy should work fine.

**Comment by nazar-pc** [comment]
Yes, normal copy works fine

**Comment by JORGETECH** [comment]
I had a similar problem copying files normally (locally, no SMB). I copied a text file and the size was changed to 60kb (original was about 500 bytes) I opened it with a Linux text editor and there were lines with more than 60000 lines on the copied text file. This is fixed in 3.1.2 beta 5 but I guess it will be included in 3.1.2 stable.

**Comment by nazar-pc** [comment]
I've tracked down that copying from SMB server to Android device works fine, however copying from Android device to SMB server gives symptoms I've described above.
